datafusion/expr-common/src/statistics.rs

```diff
@@ -857,6 +857,143 @@ pub fn compute_variance(
     ScalarValue::try_from(target_type)
 }
 
+/// Merges two distributions into a single distribution that represents their combined statistics.
+/// This creates a more general distribution that approximates the mixture of the input distributions.
+pub fn merge_distributions(a: &Distribution, b: &Distribution) -> Result<Distribution> {
+    let range_a = a.range()?;
+    let range_b = b.range()?;
+
+    // Determine data type and create combined range
+    let combined_range = if range_a.is_unbounded() || range_b.is_unbounded() {
```

Comment:
I think we could use `Interval::union` here: https://docs.rs/datafusion/latest/datafusion/logical_expr/interval_arithmetic/struct.Interval.html#method.union

Reply:
Great, I found the `Interval::union` works with intervals of the same data type.

I seems that we can loose the requirement, such as, `Int64` with `Int32`, `int` with `float`, etc also can be unioned.